### PR TITLE
add ldscsim

### PR DESCRIPTION
**Pull request by nikbaya**
ldscsim is a simulation framework for testing LD score regression

### CODE REVIEW

File: hail/python/hail/experimental/ldscsim.py [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+    mt1 = agg_fields(mt, coef_dict=cov_coef_dict, regex=cov_regex, axis='cols')
+    return mt1.annotate_cols(__y_w_popstrat = mt1.__y+mt1.__agg_cov)
+
+@typecheck(mt = MatrixTable, 
+           str_expr=str)
+def clean_fields(mt, str_expr):
+    '''Removes fields with names that have str_expr in them'''
+    all_fields = list(mt.col)+list(mt.row)+list(mt.entry)+list(mt.globals)
+    for field_to_drop in [x for x in all_fields if str_expr in x]:
+        mt = mt.drop(field_to_drop)
```
Comment (tpoterba) [inline hail/python/hail/experimental/ldscsim.py]: this will be a bit inefficient -- can just be 

```
mt = mt.drop(*(x for x in all_fields if str_expr in x))
```

File: hail/python/hail/experimental/ldscsim.py [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+    mt5 = add_sim_description(mt=mt4,h2=h2,starttime=starttime,stoptime=stoptime,
+                              runtime=runtime,pi=pi,is_annot_inf=is_annot_inf,
+                              annot_coef_dict=annot_coef_dict,annot_regex=annot_regex,
+                              h2_normalize=h2_normalize,is_popstrat=is_popstrat,
+                              cov_coef_dict=cov_coef_dict,cov_regex=cov_regex,
+                              path_to_save=path_to_save)
+    print('\rFinished simulation! (runtime={} min)'.format(round((runtime.total_seconds())/60, 4)).ljust(100))
+    if path_to_save is not None:
+        print(f'\rWriting simulation to: {path_to_save}')
+        mt5.write(path_to_save)
```
Comment (tpoterba) [inline hail/python/hail/experimental/ldscsim.py]: This can be `mt5 = mt5.checkpoint(path_to_save)`

which does a write/read

File: hail/python/hail/experimental/ldscsim.py [hunk 3]
```diff
@@ -0,0 +1,421 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Feb 15 08:25:59 2019
+
+ldsc simulation framework
+
+@author: nbaya
+"""
+
```
Comment (tpoterba) [inline hail/python/hail/experimental/ldscsim.py]: You'll need to add something to hl.experimental.__init__.py to import and expose functions you want to be available at that level.

Probably just 'simulate' which we should rename 'simulate_phenotypes' to be clear at that level.

It also is probably worth writing a full docstring for all the parameters, and providing some notes on what it does. But you can do that in a separate PR if you want.

Comment (nikbaya) [reply hail/python/hail/experimental/ldscsim.py]: There are a couple of functions related to variant annotations and the annotation-informed model which I would also like to make available. Should I import them from this module or should I create a new module just for these methods?

Comment (tpoterba) [reply hail/python/hail/experimental/ldscsim.py]: A python file behaves like a module -- You can do `hl.experimental.ldscsim.foo` for instance. You should put at most one or two functions in the upper `experimental` namespace, but you can have whatever you want here.

Comment (nikbaya) [reply hail/python/hail/experimental/ldscsim.py]: okay, great
